automation/qbft/runner/scenario.go

```diff
@@ -0,0 +1,50 @@
+package runner
+
+import (
+	"context"
+
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/ibft/storage"
+	p2pv1 "github.com/bloxapp/ssv/network/p2p"
+	"github.com/bloxapp/ssv/storage/basedb"
+)
+
+// ScenarioFactory creates Scenario instances.
+type ScenarioFactory func(name string) Scenario
+
+// ScenarioContext is the context object that is passed in execution.
+type ScenarioContext struct {
+	Ctx         context.Context
+	LocalNet    *p2pv1.LocalNet
+	Stores      []*storage.QBFTStores
+	KeyManagers []spectypes.KeyManager
+	DBs         []basedb.IDb
+}
+
+// Bootstrapper bootstraps the given scenario.
+type Bootstrapper func(ctx context.Context, logger *zap.Logger, scenario Scenario) (*ScenarioContext, error)
+
+type scenarioCfg interface {
+	// NumOfOperators returns the desired number of operators for the test.
+	NumOfOperators() int
+	// NumOfBootnodes returns the desired number of bootnodes for the test.
+	// zero in case we want mdns
+	NumOfBootnodes() int
+	// NumOfFullNodes returns the desired number of full nodes for the test.
+	NumOfFullNodes() int
+}
+
+// Scenario represents a testplan for a specific scenario
+type Scenario interface {
```

Comment:
We should have a cleaner interface, e.g. following spec tests with
`Name() string` and `Run(t *testing.T)`